File: code-examples/pymongo/indexes/unique.py
# Create unique index
from pymongo import MongoClient 

# Replace the following with your MongoDB deployment's connection string.
uri = "mongodb+srv://<user>:<password>@<cluster-url>?writeConcern=majority" 

# Create a client
client = MongoClient(uri)

try:
    # Begin index
    database = client["sample_mflix"]
    movies = database["movies"]

    # Create a unique index on the "theaterId" field in the "theaters" collection.
    result = movies.create_index("theaterId", unique=True)
    print(f"Index created: {result}")
    # End index

# Close the database connection
finally:
    client.close()

File: code-examples/java-rs/indexes/unique.java
import com.mongodb.reactivestreams.client.MongoClients;
import com.mongodb.reactivestreams.client.MongoClient;
import com.mongodb.reactivestreams.client.MongoDatabase;
import com.mongodb.reactivestreams.client.MongoCollection;
import com.mongodb.client.model.IndexOptions;
import com.mongodb.client.model.Indexes;
import org.bson.Document;

import java.util.concurrent.ExecutionException;

public class CreateUniqueIndex {

    public static void main(String[] args) {
        MongoClient mongoClient = MongoClients.create("mongodb+srv://<user>:<password>@<cluster-url>?writeConcern=majority");

        MongoDatabase database = mongoClient.getDatabase("sample_mflix");
        MongoCollection<Document> collection = database.getCollection("movies");

        collection.createIndex(Indexes.ascending("theaterId"), new IndexOptions().unique(true)).subscribe(result -> {
            System.out.println("Index created: " + result);
            mongoClient.close();
        }, throwable -> {
            System.err.println("Error: " + throwable.getMessage());
            mongoClient.close();
        });
    }
}
